<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Data Exploration | Statistics for Genomics: RNAseq</title>
  <meta name="description" content="Description about Course/Book." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Data Exploration | Statistics for Genomics: RNAseq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Description about Course/Book." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Data Exploration | Statistics for Genomics: RNAseq" />
  
  <meta name="twitter:description" content="Description about Course/Book." />
  



<meta name="date" content="2025-02-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/anvil_favicon.ico" type="image/x-icon" />
<link rel="prev" href="getting-started.html"/>
<link rel="next" href="help.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/style.css" type="text/css" />
<link rel="stylesheet" href="assets/GDSCN_style/gdscn.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://www.gdscn.org/" target="_blank"><img src="assets/GDSCN_style/logo-gdscn.png" style="width: 80%; padding-left: 15px; padding-top: 8px;"</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Book</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#skills-level"><i class="fa fa-check"></i>Skills Level</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#gdscn-collection"><i class="fa fa-check"></i>GDSCN Collection</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="overview-and-learning-objectives.html"><a href="overview-and-learning-objectives.html"><i class="fa fa-check"></i><b>1</b> Overview and Learning Objectives</a>
<ul>
<li class="chapter" data-level="1.1" data-path="overview-and-learning-objectives.html"><a href="overview-and-learning-objectives.html#activity-context"><i class="fa fa-check"></i><b>1.1</b> Activity Context</a></li>
<li class="chapter" data-level="1.2" data-path="overview-and-learning-objectives.html"><a href="overview-and-learning-objectives.html#learning-objectives"><i class="fa fa-check"></i><b>1.2</b> Learning Objectives</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="lesson-plan.html"><a href="lesson-plan.html"><i class="fa fa-check"></i><b>2</b> Lesson Plan</a></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting Started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started.html"><a href="getting-started.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started.html"><a href="getting-started.html#step-1-create-a-google-account"><i class="fa fa-check"></i><b>3.2</b> Step 1: Create a Google Account</a></li>
<li class="chapter" data-level="3.3" data-path="getting-started.html"><a href="getting-started.html#step-2-sign-in-to-terra"><i class="fa fa-check"></i><b>3.3</b> Step 2: Sign in to Terra</a></li>
<li class="chapter" data-level="3.4" data-path="getting-started.html"><a href="getting-started.html#step-3-share-your-id"><i class="fa fa-check"></i><b>3.4</b> Step 3: Share your ID</a></li>
</ul></li>
<li class="part"><span><b>RNASEQ PREPARATION</b></span></li>
<li class="chapter" data-level="4" data-path="data-exploration.html"><a href="data-exploration.html"><i class="fa fa-check"></i><b>4</b> Data Exploration</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-exploration.html"><a href="data-exploration.html#about-the-data"><i class="fa fa-check"></i><b>4.1</b> About the Data</a></li>
<li class="chapter" data-level="4.2" data-path="data-exploration.html"><a href="data-exploration.html#using-geoquery"><i class="fa fa-check"></i><b>4.2</b> Using <code>GEOquery</code></a></li>
<li class="chapter" data-level="4.3" data-path="data-exploration.html"><a href="data-exploration.html#geoquery-record-types"><i class="fa fa-check"></i><b>4.3</b> <code>GEOquery</code> Record Types</a></li>
<li class="chapter" data-level="4.4" data-path="data-exploration.html"><a href="data-exploration.html#kabuki-dataset-metadata"><i class="fa fa-check"></i><b>4.4</b> Kabuki Dataset Metadata</a></li>
<li class="chapter" data-level="4.5" data-path="data-exploration.html"><a href="data-exploration.html#pull-in-counts"><i class="fa fa-check"></i><b>4.5</b> Pull in counts</a></li>
<li class="chapter" data-level="4.6" data-path="data-exploration.html"><a href="data-exploration.html#explore-in-isee"><i class="fa fa-check"></i><b>4.6</b> Explore in iSEE</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="help.html"><a href="help.html"><i class="fa fa-check"></i><b>A</b> Help</a></li>
<li class="chapter" data-level="B" data-path="give-us-feedback.html"><a href="give-us-feedback.html"><i class="fa fa-check"></i><b>B</b> Give Us Feedback</a></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/OTTR_Template" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by: </a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<a href="http://jhudatascience.org/"><img src="https://jhudatascience.org/images/dasl.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
<p style="padding-left: 40px;"><div class="trapezoid" style = "padding-left: 40px;"><span><a href="https://docs.google.com/forms/d/e/1FAIpQLScrDVb_utm55pmb_SHx-RgELTEbCCWdLea0T3IzS0Oj00GE4w/viewform?usp=pp_url&entry.1565230805=Statistics+for+Genomics:+RNAseq"> Click here to provide feedback</a> <img src="assets/itcr_arrow.png" style=" width: 10%" ></span></div></p>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistics for Genomics: RNAseq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=10.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
</head>



<div class="hero-image-container">
  <img class= "hero-image" src="assets/GDSCN_style/dasl_thin_main_image.png">
</div>
<div id="data-exploration" class="section level1" number="4">
<h1><span class="header-section-number">Chapter 4</span> Data Exploration</h1>
<div id="about-the-data" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> About the Data</h2>
<p>Mendelian disorders of the epigenetic machinery (MDEMs) are a relatively new group of multiple congenital anomaly and intellectual disability syndromes. These disorders result from mutations in genes responsible for epigenetic machinery. In other words, genes responsible for controlling the <a href="https://youtu.be/_aAhcNjmvhc">epigenome</a> lose their normal function.</p>
<p>Despite having different causative genes, these disorders share similarities in disease presentation. This physical similarity, or phenotypic convergence, could be due to these mutations causing similar effects at the epigenomic level. Such epigenetic changes then lead to similarities gene expression.</p>
<p>Scientists at Johns Hopkins designed an experiment to identify abnormalities shared across multiple MDEMs, in order to causally relate epigenetic variation to disease phenotypes. As part of this experiment, scientists examined gene expression (RNA-Seq) states from mouse models of three MDEMs (Kabuki types 1&amp;2 and Rubinstein-Taybi syndromes).</p>
<p>You can find out more about this experiment <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE162176">here</a>.</p>
<p><img src="04-data_files/figure-html/1Xd-ZdWF-di4RnQAgF5a6J9fW7eIZtH29QYR7Req3Krc_g11e8aba06f7_0_6.png" title="Cartoon of epigenetic machinery, including a writer, reader, and eraser interacting with a histone. The histones, nucleosomes, and chromatin are all depicted as coiling into a chromosome." alt="Cartoon of epigenetic machinery, including a writer, reader, and eraser interacting with a histone. The histones, nucleosomes, and chromatin are all depicted as coiling into a chromosome." width="480" /></p>
</div>
<div id="using-geoquery" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Using <code>GEOquery</code></h2>
<p>The <a href="https://www.ncbi.nlm.nih.gov/geo/">NCBI Gene Expression Omnibus (GEO)</a> is an international public repository that archives and freely distributes microarray, next-generation sequencing, and other forms of high-throughput functional genomics data submitted by the research community. We will use the Bioconductor package <a href="https://bioconductor.org/packages/release/bioc/html/GEOquery.html"><code>GEOquery</code></a> to load data from GEO.</p>
<p>Using <code>GEOquery</code> is convenient because it allows us to get the data programmatically without having to download anything manually. This ensures anyone following in our footsteps can follow what we did exactly.</p>
<p>First, install <code>GEOquery</code> and load the library using the following code. if you are asked to update packages, you can type ‘n’ for ‘no’.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="data-exploration.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Install and load GEOquery</span></span>
<span id="cb1-2"><a href="data-exploration.html#cb1-2" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;GEOquery&quot;</span>)</span>
<span id="cb1-3"><a href="data-exploration.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span></code></pre></div>
</div>
<div id="geoquery-record-types" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> <code>GEOquery</code> Record Types</h2>
<p>GEO contains <a href="https://www.ncbi.nlm.nih.gov/geo/info/overview.html">several different record types</a>. The most straightforward is a ‘Sample’ record. A Sample record describes the conditions under which an individual Sample was handled, the manipulations it underwent, and the abundance measurement of each element derived from it. Each Sample record is assigned a unique and stable GEO accession number (GSMxxx). A Sample entity may be included in multiple ‘Series’.</p>
<p>a ‘Series’ record defines a set of related Samples considered to be part of a group. This record describes how the Samples are related and provides information about the experiment. Series records may also contain tables describing extracted data, summary conclusions, or analyses. Each Series record is assigned a unique and stable GEO accession number (GSExxx).</p>
<p>We need to locate the correct ‘Series’ number for this experiment.</p>
<p><img src="04-data_files/figure-html/1Xd-ZdWF-di4RnQAgF5a6J9fW7eIZtH29QYR7Req3Krc_gcf1264c749_0_140.png" title="Screenshot of the GEO accession page corresponding to the dataset we are looking for. The Series record number is highlighted." alt="Screenshot of the GEO accession page corresponding to the dataset we are looking for. The Series record number is highlighted." width="480" /></p>
</div>
<div id="kabuki-dataset-metadata" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Kabuki Dataset Metadata</h2>
<p>We will use the <code>getGEO()</code> function to locate the experiment’s data using the Series record number.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="data-exploration.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Indicate which Series to download</span></span>
<span id="cb2-2"><a href="data-exploration.html#cb2-2" aria-hidden="true" tabindex="-1"></a>gse <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="st">&quot;GSE162176&quot;</span>)</span></code></pre></div>
<p>The output from <code>getGEO()</code> on a Series record type is a list of objects called an <code>ExpressionSet</code>. In our case, there is only one ExpressionSet, so we can select the first item (the data) from the list using brackets.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="data-exploration.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the first item in the list</span></span>
<span id="cb3-2"><a href="data-exploration.html#cb3-2" aria-hidden="true" tabindex="-1"></a>exp_set <span class="ot">&lt;-</span> gse[[<span class="dv">1</span>]]</span></code></pre></div>
<p>We can look at the ExpressionSet metadata using <code>pData()</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="data-exploration.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract phenotypic data</span></span>
<span id="cb4-2"><a href="data-exploration.html#cb4-2" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="ot">&lt;-</span> <span class="fu">pData</span>(exp_set)</span>
<span id="cb4-3"><a href="data-exploration.html#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pheno_data)</span></code></pre></div>
<pre><code>##  [1] &quot;title&quot;                   &quot;geo_accession&quot;          
##  [3] &quot;status&quot;                  &quot;submission_date&quot;        
##  [5] &quot;last_update_date&quot;        &quot;type&quot;                   
##  [7] &quot;channel_count&quot;           &quot;source_name_ch1&quot;        
##  [9] &quot;organism_ch1&quot;            &quot;characteristics_ch1&quot;    
## [11] &quot;characteristics_ch1.1&quot;   &quot;characteristics_ch1.2&quot;  
## [13] &quot;characteristics_ch1.3&quot;   &quot;molecule_ch1&quot;           
## [15] &quot;extract_protocol_ch1&quot;    &quot;extract_protocol_ch1.1&quot; 
## [17] &quot;taxid_ch1&quot;               &quot;description&quot;            
## [19] &quot;data_processing&quot;         &quot;data_processing.1&quot;      
## [21] &quot;data_processing.2&quot;       &quot;platform_id&quot;            
## [23] &quot;contact_name&quot;            &quot;contact_email&quot;          
## [25] &quot;contact_laboratory&quot;      &quot;contact_department&quot;     
## [27] &quot;contact_institute&quot;       &quot;contact_address&quot;        
## [29] &quot;contact_city&quot;            &quot;contact_state&quot;          
## [31] &quot;contact_zip/postal_code&quot; &quot;contact_country&quot;        
## [33] &quot;data_row_count&quot;          &quot;instrument_model&quot;       
## [35] &quot;library_selection&quot;       &quot;library_source&quot;         
## [37] &quot;library_strategy&quot;        &quot;relation&quot;               
## [39] &quot;relation.1&quot;              &quot;supplementary_file_1&quot;   
## [41] &quot;age (approximate):ch1&quot;   &quot;cell type:ch1&quot;          
## [43] &quot;disease state:ch1&quot;       &quot;strain:ch1&quot;</code></pre>
<p>View the data to see which Samples are contained in this ExpressionSet.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="data-exploration.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Explore the metadata</span></span>
<span id="cb6-2"><a href="data-exploration.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(pheno_data)</span></code></pre></div>
<pre><code>##                                   title geo_accession                status
## GSM4944267 Kabuki syndrome type 1 rep 1    GSM4944267 Public on Aug 31 2021
## GSM4944268 Kabuki syndrome type 1 rep 2    GSM4944268 Public on Aug 31 2021
## GSM4944269 Kabuki syndrome type 1 rep 3    GSM4944269 Public on Aug 31 2021
## GSM4944270 Kabuki syndrome type 1 rep 4    GSM4944270 Public on Aug 31 2021
## GSM4944271 Kabuki syndrome type 1 rep 5    GSM4944271 Public on Aug 31 2021
##            submission_date last_update_date type
## GSM4944267     Nov 25 2020      Aug 31 2021  SRA
## GSM4944268     Nov 25 2020      Aug 31 2021  SRA
## GSM4944269     Nov 25 2020      Aug 31 2021  SRA
## GSM4944270     Nov 25 2020      Aug 31 2021  SRA
## GSM4944271     Nov 25 2020      Aug 31 2021  SRA</code></pre>
<p>The GEO record provides information about each of the samples (aka “metadata”). You can use the <code>table()</code> function to tabulate how many samples there are for each of the three disease states (along with their corresponding wild type controls).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="data-exploration.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>( pheno_data<span class="sc">$</span><span class="st">`</span><span class="at">disease state:ch1</span><span class="st">`</span> )</span></code></pre></div>
<pre><code>## 
##              Kabuki syndrome type 1              Kabuki syndrome type 2 
##                                   5                                   5 
##    Rubinstein-Taybi syndrome type 1          Wild-type (Kabuki cohorts) 
##                                   5                                   5 
## Wild-type (Rubinstein-Taybi cohort) 
##                                   7</code></pre>
<div class="fyi">
<p>QUESTIONS:</p>
<ol style="list-style-type: decimal">
<li>What was the cell type (<code>cell type:ch1</code>) used in this experiment?</li>
<li>Which column contains information about the ages?</li>
<li>Which age has the most samples?</li>
</ol>
</div>
</div>
<div id="pull-in-counts" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Pull in counts</h2>
<p>Retrieving GSE162176 using <code>getGEO()</code> currently does not obtain RNA-seq expression data. We have cached a <code>SummarizedExperiment</code> object in the <a href="https://anvil.terra.bio/#workspaces/gdscn-exercises/datasets">GDSCN datasets</a> Workspace. This object contains metadata, counts, and abundance information as produced by the <a href="https://nf-co.re/rnaseq/3.6">nf-co.re/rnaseq/3.6</a> pipeline using the <a href="https://www.gencodegenes.org/mouse/release_M23.html">GENCODE M23</a> annotation.</p>
<p><img src="04-data_files/figure-html/1Xd-ZdWF-di4RnQAgF5a6J9fW7eIZtH29QYR7Req3Krc_g12145d7e666_0_0.png" title="Screenshot of the GDSCN datasets Workspace showing the cached GSE162176.rds dataset." alt="Screenshot of the GDSCN datasets Workspace showing the cached GSE162176.rds dataset." width="480" /></p>
<p>The Bioconductor <a href="http://bioconductor.org/packages/AnVIL">AnVIL package</a> provides a <code>gsutil_cp()</code> function to streamline transfers between AnVIL Workspaces. Note that the <a href="https://cloud.google.com/storage/docs/gsutil#syntax">syntax for accessing Google Cloud Storage resources</a> is <code>gs://BUCKET_NAME/OBJET_NAME</code>, and that each AnVIL Workspace has an associated bucket. The following command transfers the <code>GSE162176.rds</code> file from the GDSCN datasets Workspace into your Workspace.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="data-exploration.html#cb10-1" aria-hidden="true" tabindex="-1"></a>AnVIL<span class="sc">::</span><span class="fu">gsutil_cp</span>( <span class="st">&quot;gs://fc-8529d29f-ac62-4c10-9f01-14f4d7612ae0/GSE162176.rds&quot;</span>, <span class="st">&quot;.&quot;</span> )</span></code></pre></div>
<p>Once you have obtained a copy of the <code>.rds</code> file, load it into your environment using the <code>readRDS()</code> function. Note that you will need to install the <a href="http://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a> package if you have not previously done so.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="data-exploration.html#cb11-1" aria-hidden="true" tabindex="-1"></a>AnVIL<span class="sc">::</span><span class="fu">install</span>( <span class="st">&quot;SummarizedExperiment&quot;</span> )</span>
<span id="cb11-2"><a href="data-exploration.html#cb11-2" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>( <span class="st">&quot;GSE162176.rds&quot;</span> )</span></code></pre></div>
<p>Confirm that you have successfully loaded the data by creating a scatterplot comparing a Kabuki sample and a Wild-type sample.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="data-exploration.html#cb12-1" aria-hidden="true" tabindex="-1"></a>kabuki <span class="ot">&lt;-</span> <span class="fu">assay</span>(se)[,<span class="st">&quot;SRX9584943&quot;</span>]</span>
<span id="cb12-2"><a href="data-exploration.html#cb12-2" aria-hidden="true" tabindex="-1"></a>wildtype <span class="ot">&lt;-</span> <span class="fu">assay</span>(se)[,<span class="st">&quot;SRX9584958&quot;</span>]</span>
<span id="cb12-3"><a href="data-exploration.html#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>( <span class="fu">log2</span>(kabuki<span class="sc">+</span><span class="dv">1</span>), <span class="fu">log2</span>(wildtype<span class="sc">+</span><span class="dv">1</span>) )</span></code></pre></div>
<p><img src="04-data_files/figure-html/1Xd-ZdWF-di4RnQAgF5a6J9fW7eIZtH29QYR7Req3Krc_g12145d7e666_0_9.png" title="Screenshot of RStudio with an R Notebook that creates a scatterplot comparing kabuki and wildtype log2 counts." alt="Screenshot of RStudio with an R Notebook that creates a scatterplot comparing kabuki and wildtype log2 counts." width="480" /></p>
<div class="fyi">
<p>QUESTIONS:</p>
<ol start="4" style="list-style-type: decimal">
<li>Which SRX accession numbers correspond to Rubinstein-Taybi?</li>
<li>Create a scatterplot comparing a Rubinstein-Taybi sample and a Wild-type sample</li>
</ol>
</div>
</div>
<div id="explore-in-isee" class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> Explore in iSEE</h2>
<p>The Bioconductor <a href="https://bioconductor.org/packages/iSEE">Interactive SummarizedExperiment Explorer (iSEE)</a> provides an interactive Shiny-based graphical user interface for exploring data stored in SummarizedExperiment objects and it’s extensions such as SingleCellExperiment. One feature that we will use here is the ability to explore what experimental conditions are present in this dataset and how gene expression changes between conditions.</p>
<p><img src="04-data_files/figure-html/1Xd-ZdWF-di4RnQAgF5a6J9fW7eIZtH29QYR7Req3Krc_g12145d7e666_0_14.png" title="Screenshot of an iSEE session with three panels: 1) Column data table showing sample metadata, 2) Column data plot showing sample titles vs disease, and 3) Feature assay plot showing gene counts across disease states." alt="Screenshot of an iSEE session with three panels: 1) Column data table showing sample metadata, 2) Column data plot showing sample titles vs disease, and 3) Feature assay plot showing gene counts across disease states." width="480" /></p>
<p>You can launch an interactive explorer simply by calling the <code>iSEE()</code> function with only a SummarizedExperiment object. However, you can also configure the explorer by passing additional parameters such as the types of panels that you wish to appear. Here we launch iSEE with panels allowing exploration of the sample metadata as a table, the sample metadata as a plot, and the gene expression data as a plot. Note that you will need to install the iSEE package if you have not previously done so.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="data-exploration.html#cb13-1" aria-hidden="true" tabindex="-1"></a>AnVIL<span class="sc">::</span><span class="fu">install</span>( <span class="st">&quot;iSEE&quot;</span> )</span>
<span id="cb13-2"><a href="data-exploration.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>( <span class="st">&quot;iSEE&quot;</span> )</span>
<span id="cb13-3"><a href="data-exploration.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">iSEE</span>( se, <span class="fu">list</span>( <span class="fu">ColumnDataTable</span>(), <span class="fu">ColumnDataPlot</span>(), <span class="fu">FeatureAssayPlot</span>() ) )</span></code></pre></div>
<div class="fyi">
<p>QUESTIONS:</p>
<ol start="6" style="list-style-type: decimal">
<li>In the “Column data plot” panel, create a plot with age on the y-axis and disease on the x-axis. Which disease states have the most age diversity? The least?</li>
<li>In the “Feature assay plot” panel, create a plot with ENSMUSG00000000001.4 on the y-axis and disease on the x-axis. What conclusions can you make if you plot counts? Does it change if you plot abundance?</li>
</ol>
</div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="data-exploration.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.3 (2022-03-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.5 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] GEOquery_2.62.2     Biobase_2.54.0      BiocGenerics_0.40.0
## 
## loaded via a namespace (and not attached):
##  [1] bslib_0.3.1         compiler_4.1.3      pillar_1.7.0       
##  [4] BiocManager_1.30.16 jquerylib_0.1.4     highr_0.9          
##  [7] R.methodsS3_1.8.1   R.utils_2.12.2      tools_4.1.3        
## [10] digest_0.6.29       jsonlite_1.8.0      evaluate_0.15      
## [13] lifecycle_1.0.1     tibble_3.1.6        pkgconfig_2.0.3    
## [16] png_0.1-7           rlang_1.0.2         DBI_1.1.2          
## [19] cli_3.2.0           curl_4.3.2          yaml_2.3.5         
## [22] xfun_0.26           fastmap_1.1.0       xml2_1.3.3         
## [25] dplyr_1.0.8         stringr_1.4.0       httr_1.4.2         
## [28] knitr_1.33          generics_0.1.2      fs_1.5.2           
## [31] vctrs_0.4.1         sass_0.4.1          hms_1.1.1          
## [34] tidyselect_1.1.2    data.table_1.14.2   glue_1.6.2         
## [37] R6_2.5.1            fansi_1.0.3         ottrpal_1.0.1      
## [40] rmarkdown_2.10      bookdown_0.24       limma_3.50.3       
## [43] tidyr_1.2.0         purrr_0.3.4         readr_2.1.2        
## [46] tzdb_0.3.0          magrittr_2.0.3      ellipsis_0.3.2     
## [49] htmltools_0.5.2     assertthat_0.2.1    utf8_1.2.2         
## [52] stringi_1.7.6       crayon_1.5.1        R.oo_1.24.0</code></pre>

</div>
</div>



<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="getting-started.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="help.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"edit": "https://github.com/jhudsl/GDSCN_Book_Statistics_for_Genomics_RNA-seq/edit/main/%s"
}
});
});
</script>

</body>

</html>
